### ISSUE TITLE
Build Error in Ubuntu 16.04 with 1.11.33.2-beta

### ISSUE THREAD

**Issue by ricardogomes** [pagespeed/ngx_pagespeed#1202]
Hi

I'm new to building nginx with pagespeed by hand, but I think I followed this [article on DigitalOcean](https://www.digitalocean.com/community/tutorials/how-to-add-ngx_pagespeed-to-nginx-on-ubuntu-14-04) OK (changing the version numbers).

nginx-1.10.0 in Ubuntu 16.04

I'm getting this output when running `sudo dpkg-buildpackage -b`

```
...
o objs/addon/src/ngx_pagespeed.o \
	/home/ricardo/custom-nginx/nginx-1.10.0/debian/modules/ngx_pagespeed/src/ngx_pagespeed.cc
cc: internal compiler error: Killed (program cc1plus)
...
objs/Makefile:1317: recipe for target 'objs/addon/src/ngx_pagespeed.o' failed
make[3]: *** [objs/addon/src/ngx_pagespeed.o] Error 4
make[3]: Leaving directory '/home/ricardo/custom-nginx/nginx-1.10.0/debian/build-light'
Makefile:8: recipe for target 'build' failed
make[2]: *** [build] Error 2
make[2]: Leaving directory '/home/ricardo/custom-nginx/nginx-1.10.0/debian/build-light'
debian/rules:161: recipe for target 'build.arch.light' failed
make[1]: *** [build.arch.light] Error 2
make[1]: Leaving directory '/home/ricardo/custom-nginx/nginx-1.10.0'
debian/rules:143: recipe for target 'build' failed
make: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2
```

My rules file:

```
...
light_configure_flags := \
    $(common_configure_flags) \
    --with-http_gzip_static_module \
    --without-http_browser_module \
    --without-http_geo_module \
    --without-http_limit_req_module \
    --without-http_limit_conn_module \
    --without-http_memcached_module \
    --without-http_referer_module \
    --without-http_scgi_module \
    --without-http_split_clients_module \
    --without-http_ssi_module \
    --without-http_userid_module \
    --without-http_uwsgi_module \
    --add-module=$(MODULESDIR)/nginx-echo \
    --add-module=$(MODULESDIR)/ngx_pagespeed
```

I try it in the full mode as well, same error.

What am I missing?

Thanks

**Comment by crowell** [comment]
how much memory do you have on your VM? It looks to me like your compiler OOM'd from the line

`cc: internal compiler error: Killed (program cc1plus)`

**Comment by ricardogomes** [comment]
512MB

Never thought about that. 

What would be the minimum RAM need to build this?

Thanks

**Comment by crowell** [comment]
To be totally honest, I'm not exactly sure how much memory is needed. I _think_ 512mb should be workable if there's nothing else competing for the ram. Can you try building immediately after a reboot?

**Comment by ricardogomes** [comment]
@crowell  Thanks, it was the RAM

I started a 4GB (just in case) VM in DigitalOcean, and everything built. Copied over the files to my original VM and they installed perfectly.

**Comment by TihomirS** [comment]
Hi Ricardo,
I'm trying to do the same .. and same error :)
Did you try to add swap file before ram update?
Can you post your used psol version? 

Thanks

**Comment by ricardogomes** [comment]
Hi @TihomirS 

No, I didn't. Since DigitalOcean is SSD based they advise you not to enable swap. But it didn't cost more than a few cents, to run the 4GB VM, I destroyed it afterwards. 

The versions where

nginx-1.10.0
ngx_pagespeed release-1.11.33.2-beta.zip
psol 1.11.33.2.tar.gz

**Comment by TihomirS** [comment]
Hi @ricardogomes 
thank you for fast reply .. will try again with 3g swap (only for build) and this versions :)

and yes .. if fail again will try with 4 g real ram ..
best wishes from Bulgaria
T.

**Comment by TihomirS** [comment]
WOW :) with right psol 1.11.33.2.tar.gz build is smooth and OK
Thank you @ricardogomes

**Comment by pra85** [comment]
I faced a similar issue when building pagespeed v1.12.34.2-stable with nginx. I was doing this on an Amazon LightSail VPS (with 512 MB Memory). I re-tried a few times and but there always build failure ( The error message was `cc: internal compiler error: Killed (program cc1plus)` but the files causing it changed) But then I tried with a 1GB Memory instance (EC2 t2.micro) and was able to build without problem. Didn't test extensively,  but I suppose we need around 1GB Memory to make this work
